src/Core/Checkout/Promotion/Cart/PromotionCollector.php

```diff
@@ -76,85 +104,104 @@ public function collect(CartDataCollection $data, Cart $original, SalesChannelCo
             // However, this cannot be applied directly, because it does not yet have any items in the cart.
             // Therefore the code is stored in the extension and as soon
             // as the user has enough items in the cart, it is added again.
-            $cartExtension = $original->getExtension(CartExtension::KEY);
-            if (!$cartExtension instanceof CartExtension) {
-                $cartExtension = new CartExtension();
-                $original->addExtension(CartExtension::KEY, $cartExtension);
-            }
+            $cartExtension = $original->getExtensionOfType(CartExtension::KEY, CartExtension::class);
+            $cartExtension ??= new CartExtension();
```

Comment:
```suggestion
            $cartExtension = $original->getExtensionOfType(CartExtension::KEY, CartExtension::class) ?? new CartExtension();
```

This is more the standard here isn't it? `??=` is mostly used when it's not possible to do `??` (variable passed in as argument mostly)